zeppelin-cluster/yarn/src/test/java/org/apache/zeppelin/cluster/yarn/YarnClientTest.java

```diff
@@ -0,0 +1,108 @@
+    yarnConfiguration.set(YarnConfiguration.YARN_MINICLUSTER_USE_RPC, "true");
+
+    miniYARNCluster = new MiniYARNCluster(YarnClientTest.class.getName(), 1, 1, 1);
+    miniYARNCluster.init(yarnConfiguration);
+    miniYARNCluster.start();
+
+    final CountDownLatch waitUntilLaunched = new CountDownLatch(1);
+    final AtomicInteger checkerRunning = new AtomicInteger(0);
+
+    new Thread(new Runnable() {
+      @Override
+      public void run() {
+        while (0 < checkerRunning.get()) {
+          if (!"0".equals(miniYARNCluster.getConfig().get(YarnConfiguration.RM_ADDRESS).split(":")[1])) {
+            waitUntilLaunched.countDown();
+          }
+        }
+      }
+    }).start();
+
+    try {
+      waitUntilLaunched.await(10, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      // Do nothing
+    }
+
+    if ("0".equals(miniYARNCluster.getConfig().get(YarnConfiguration.RM_ADDRESS).split(":")[1])) {
+      throw new RuntimeException("Cannot run MiniYarnCluster for testing yarn");
+    }
+  }
+
+  @AfterClass
+  public static void shutdownCluster() {
+    try {
+      miniYARNCluster.close();
+    } catch (IOException e) {
+      // Do nothing
+    }
+  }
+
+  @Test
+  public void clientTest() throws Exception {
+    System.out.println("RM_ADDRESS = " + miniYARNCluster.getConfig().get(YarnConfiguration.RM_ADDRESS));
+    Client client = new Client(miniYARNCluster.getConfig());
+    client.start();
+    RemoteInterpreterProcess remoteInterpreterProcess = client.createInterpreter("id", "name", "fake", Maps.<String, String>newHashMap(), new Properties(), 100000, null, null, "fakeHome", "fake");
+    String parentClasspath = Paths.get(YarnRemoteInterpreterServer.class.getProtectionDomain().getCodeSource().getLocation().toURI()).getParent().getParent().getParent().toAbsolutePath().toString();
+    String commonClasspath = parentClasspath + "/common/target";
+    String yarnClasspath = parentClasspath + "/yarn/target";
+    System.out.println("parentClasspath: " + parentClasspath);
+    System.out.println("commonClasspath: " + commonClasspath);
+    System.out.println("yarnClasspath: " + yarnClasspath);
+    String interpreterClasspath = Paths.get(parentClasspath).getParent().toAbsolutePath().toString() + "/zeppelin-interpreter/target";
+    ((RemoteInterpreterYarnProcess)remoteInterpreterProcess).setExtraClasspath(commonClasspath + "/classes" + ApplicationConstants.CLASS_PATH_SEPARATOR + commonClasspath + "/lib/*" + ApplicationConstants.CLASS_PATH_SEPARATOR + yarnClasspath + "/classes" + ApplicationConstants.CLASS_PATH_SEPARATOR + yarnClasspath + "/lib/*" + ApplicationConstants.CLASS_PATH_SEPARATOR + interpreterClasspath+ "/classes" + ApplicationConstants.CLASS_PATH_SEPARATOR + interpreterClasspath + "/lib/*");
+    remoteInterpreterProcess.start("user", false);
+    assertNotNull("Host should not be null", remoteInterpreterProcess.getHost());
+    assertFalse("Port should not be -1", -1 == remoteInterpreterProcess.getPort());
```

Comment:
@zjffdu The main feature of this cluster manager is to start remote process in another machine and then return those two information. I think it's enough and if we start some interpreter while testing it. it would be guaranteed some interpreters already is built before the test and it's too hard to set specific order for it.